### PR TITLE
chore: improve tofu handling

### PR DESCRIPTION
**Pull request by pivotal-marcela-campo**
### Checklist:

* [ ] Have you added or updated tests to validate the changed functionality?
* [ ] Have you added Release Notes in the docs repositories?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: pkg/providers/tf/upgrade.go [hunk 1]
```diff
@@ -104,8 +104,8 @@ func (provider *TerraformProvider) performTerraformUpgrade(ctx context.Context,
 		return err
 	}
 
-	if currentTfVersion.LessThan(version.Must(version.NewVersion("0.12.0"))) {
-		return errors.New("upgrade only supported for Terraform versions >= 0.12.0")
+	if currentTfVersion.LessThan(version.Must(version.NewVersion("1.5.0"))) {
+		return errors.New("upgrade only supported for Terraform versions >= 1.5.0")
```
Comment (blgm) [inline pkg/providers/tf/upgrade.go:108]: Should the word `Terraform` still be here?

File: pkg/providers/tf/workspace/workspace_test.go [hunk 2]
```diff
@@ -213,110 +213,6 @@ func TestTerrafromWorkspace_StateTFVersion(t *testing.T) {
 	})
 }
 
-func TestCustomTerraformExecutor012(t *testing.T) {
```
Comment (blgm) [inline pkg/providers/tf/workspace/workspace_test.go:216]: Has the implementation been deleted?

**Review by blgm — APPROVED** [review]

**Review by FelisiaM — APPROVED** [review]

File: pkg/providers/tf/workspace/workspace_test.go (hunk 2 above)
Comment (pivotal-marcela-campo) [reply pkg/providers/tf/workspace/workspace_test.go:216]: Yes, the executors were easy to delete at the beginning. Remaining logic was deleted in https://github.com/cloudfoundry/cloud-service-broker/pull/975/commits/ca1610cd04ff85332d6b9d6a49ec19960a6aef8e within this PR. It seems I did miss cleaning up a `NewInit012` in `terraform_command` but it is only invoked in tests already, has been removed from the code. Will add that bit to this PR as well.

Comment (pivotal-marcela-campo) [reply pkg/providers/tf/workspace/workspace_test.go:216]: I added commit https://github.com/cloudfoundry/cloud-service-broker/pull/975/commits/590d1147db0564892819dc9f13f1443cf518c3d0 to remove the mentioned logic. I will just merge being that it was unused.

**Pull request merged**